torch/nn/parallel/distributed.py

```diff
@@ -347,17 +347,27 @@ class DistributedDataParallel(Module, Joinable):
     To use ``DistributedDataParallel`` on a host with N GPUs, you should spawn
     up ``N`` processes, ensuring that each process exclusively works on a single
     GPU from 0 to N-1. This can be done by either setting
-    ``CUDA_VISIBLE_DEVICES`` for every process or by calling:
+    ``CUDA_VISIBLE_DEVICES`` for every process or by calling the following API for GPUs,
 
         >>> # xdoctest: +SKIP("undefined variables")
         >>> torch.cuda.set_device(i)
 
+    and XPUs,
+
+        >>> # xdoctest: +SKIP("undefined variables")
+        >>> torch.xpu.set_device(i)
+
     where i is from 0 to N-1. In each process, you should refer the following
     to construct this module:
 
         >>> # xdoctest: +SKIP("undefined variables")
+        >>> if torch.cuda.is_available():
+        >>>     vendor_backend = 'nccl'
+        >>> elif torch.xpu.is_available():
+        >>>     vendor_backend = 'xccl'
+        >>>
```

Comment:
```suggestion
        >>> if torch.accelerator.is_available():
        >>>     acc = torch.accelerator.current_accelerator()
        >>>     vendor_backend = torch.distributed.get_default_backend_for_device(device_type)
        >>>
```
Following @wconstab 's comments, we could generalize this code a bit.